src/docquery/ext/pipeline_image_classification.py

```diff
@@ -0,0 +1,200 @@
+            self.model_type = ModelType.VisionEncoderDecoder
+        else:
+            self.model_type = ModelType.Standard
+
+    def _sanitize_parameters(self, top_k=None):
+        postprocess_params = {}
+        if top_k is not None:
+            postprocess_params["top_k"] = top_k
+        return {}, {}, postprocess_params
+
+    def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Image"]], **kwargs):
+        """
+        Assign labels to the image(s) passed as inputs.
+
+        Args:
+            images (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            top_k (`int`, *optional*, defaults to 5):
+                The number of top labels that will be returned by the pipeline. If the provided number is higher than
+                the number of labels available in the model configuration, it will default to the number of labels.
+
+        Return:
+            A dictionary or a list of dictionaries containing result. If the input is a single image, will return a
+            dictionary, if the input is a list of several images, will return a list of dictionaries corresponding to
+            the images.
+
+            The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+        return super().__call__(images, **kwargs)
+
+    def preprocess(self, image):
+        image = load_image(image)
+        model_inputs = self.feature_extractor(images=image, return_tensors=self.framework)
+
+        if self.model_type == ModelType.VisionEncoderDecoder:
+            model_inputs = {
+                "inputs": model_inputs["pixel_values"],
+                "max_length": self.model.decoder.config.max_position_embeddings,
+                "decoder_input_ids": self.tokenizer(
+                    "<s_rvlcdip>",
+                    add_special_tokens=False,
+                    return_tensors=self.framework,
+                ).input_ids,
+                "return_dict_in_generate": True,
+            }
+        return model_inputs
+
+    def _forward(self, model_inputs):
+        if self.model_type == ModelType.VisionEncoderDecoder:
+            model_outputs = self.model.generate(**model_inputs)
+        else:
+            model_outputs = self.model(**model_inputs)
```

Comment:
And can replace this line with `model_outputs = super()._forward(model_inputs)`

Reply:
commented separately about why we copy/pasted.